crypto/ec/ecx_backend.c

```diff
@@ -52,25 +52,20 @@ int ossl_ecx_public_from_private(ECX_KEY *key)
     return 1;
 }
 
-int ossl_ecx_key_fromdata(ECX_KEY *ecx, const OSSL_PARAM params[],
+int ossl_ecx_key_fromdata(ECX_KEY *ecx, const OSSL_PARAM *param_pub_key,
+                          const OSSL_PARAM *param_priv_key,
                           int include_private)
 {
     size_t privkeylen = 0, pubkeylen = 0;
-    const OSSL_PARAM *param_priv_key = NULL, *param_pub_key;
     unsigned char *pubkey;
 
     if (ecx == NULL)
         return 0;
 
-    param_pub_key = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_PUB_KEY);
-    if (include_private)
-        param_priv_key =
-            OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_PRIV_KEY);
-
     if (param_pub_key == NULL && param_priv_key == NULL)
```

Comment:
This logic doesnt look quite the same to me...
In the old code
````
if (param_pub_key == NULL && param_priv_key == NULL)
````
only set param_priv_key if include_private was set... So your test is different..

To be equivalent param_priv_key is passed in with include_private=1, or param_priv_key==NULL and include_private = 0, which means you either dont need include_private to be passed anymore or the above test is not right?

Reply:
Is the suggestion here to drop one of the arguments since both `param_priv_key` and `include_private` aren't required?  I tried to not change how this was used, but it would be straightforward to drop the `include_private` argument.

As for a behaviour change, the additional check on line 73 below retains the existing behaviour I believe.  That's the only block that references the private key and it's now conditional on `include_private`.  The reason behind this is that we need to honour the users choice if they supplied both the public and private keys but only asked for the public part.  The generated param decoder doesn't know what the user wants and would fill point fields in if they are supplied.

Reply:
Discussed this and it's okay as it is.